src/libraries/Microsoft.PowerFx.Core/Public/DisplayNames/RenameDriver.cs

```diff
@@ -67,16 +76,52 @@ public string ApplyRename(string expressionText)
         public bool Find(string expressionText)
         {
             var invariantExpression = _engine.GetInvariantExpression(expressionText, _baseParameters, _culture);
-
             return invariantExpression != ApplyRename(invariantExpression);
         }
 
+        private INameResolver RenameResolverHelper(INameResolver resolver, DPath pathToRename, DName updatedName)
+        {
+            if (resolver is not ReadOnlySymbolTable rost)
+            {
+                throw new Exception("Unexpected resolver");
+            }
+
+            if (!rost.TryLookup(pathToRename[0], out NameLookupInfo nameInfo))
+            {
+                return resolver;
+            }
+
+            SymbolTable st = new SymbolTable();
+            bool updated = false;
+
+            if (nameInfo.Type.Kind == Types.DKind.OptionSet)
+            {
+                OptionSet os = (OptionSet)nameInfo.Data;
+
+                if (pathToRename.Length == 1)
+                {
+                    st.AddEntity(os, updatedName);
+                    updated = true;
+                }
+                else if (pathToRename.Length == 2)
+                {
+                    if (os.TryGetValue(pathToRename[1], out OptionSetValue osValue))
+                    {
+                        st.AddEntity(new OptionSet(os.EntityName, new SingleSourceDisplayNameProvider(new Dictionary<DName, DName> { [new DName(osValue.Option)] = updatedName })));
```

Comment:
Ah: this'll probably need to have the other logical->logical pairs in the existing option set, I think?

Reply:
Good call